tasks/post_to_queue_and_trigger_step_function/post_to_queue_and_trigger_step_function.py

```diff
@@ -52,31 +52,32 @@ def process_record(
     Central method for translating record to pass along and triggering step function.
 
     Args:
-        record: The record to post.
+        record: The record to post. See schemas/input.json under Records/items for details.
         target_queue_url: The url of the queue to post the records to.
         step_function_arn: The arn of the step function to trigger.
     """
-    new_body = translate_record_body(record["body"])
-    sqs_library.post_to_fifo_queue(target_queue_url, new_body)
-    trigger_step_function(step_function_arn)
+    body = json.loads(record["body"])
+    _BODY_VALIDATE(body)
+
+    for s3_record in body["Records"]:  # Unsure if AWS sends multiple records, but data structure supports it.
+        new_body = translate_record_body(s3_record)
+        sqs_library.post_to_fifo_queue(target_queue_url, new_body)
+        trigger_step_function(step_function_arn)
```

Comment:
Are you concerned about a race condition where we post to SQS and the trigger right away? I am just wondering if we should put a delay before triggering. Thoughts?

Reply:
Haven't seen this in testing, but that means little. Added in a short delay.